### DISCUSSION TITLE
2021 User Survey Results

### DISCUSSION THREAD

**Discussion by jeremystretch** [netbox-community/netbox#6086] [General]
The results from our 2021 NetBox user survey are in! I and the other maintainers want to thank everyone who took the time to participate. Your feedback helps form the direction of NetBox's development over the coming months and years. Here are the results.

### Structured Responses

![Forms response chart. Question title: Which best describes your organization's industry?. Number of responses: 275 responses.](https://user-images.githubusercontent.com/13487278/113518896-1c095880-9557-11eb-823c-51f8fbea095b.png)

Similar to past years, we see a heavy skew toward information services, with healthy representation in education and government as well. However, that 20% "other" chunk suggests we might need to revisit the listed industry classifications.

![Forms response chart. Question title: How large is your organization (head count)?. Number of responses: 275 responses.](https://user-images.githubusercontent.com/13487278/113518917-3d6a4480-9557-11eb-8334-5993fefc1264.png)

Like last year, we've got a fairly even spread across organizations of all sizes, which is great to see.

![Forms response chart. Question title: How many NetBox users does your organization have?. Number of responses: 275 responses.](https://user-images.githubusercontent.com/13487278/113518926-4824d980-9557-11eb-97b5-bb87101fafc5.png)

![Forms response chart. Question title: Roughly how long have you been using NetBox?. Number of responses: 275 responses.](https://user-images.githubusercontent.com/13487278/113518932-5115ab00-9557-11eb-8dbc-ec3ced11d521.png)

For this year's survey, we decided to collect some information on how many of different types of objects each respondent has defined in their NetBox instance. These help inform some behind-the-scenes modeling decisions.

![Forms response chart. Question title: How many regions exist in your NetBox installation?. Number of responses: 263 responses.](https://user-images.githubusercontent.com/13487278/113518941-5b37a980-9557-11eb-813e-3f5e3a6b324a.png)

![Forms response chart. Question title: How many sites exist in your NetBox installation?. Number of responses: 266 responses.](https://user-images.githubusercontent.com/13487278/113518943-5ffc5d80-9557-11eb-9ff3-d48c2c6f5daa.png)

![Forms response chart. Question title: How many rack groups exist in your NetBox installation?. Number of responses: 263 responses.](https://user-images.githubusercontent.com/13487278/113518947-65f23e80-9557-11eb-9ee0-0148aa3dd93b.png)

![Forms response chart. Question title: How many racks exist in your NetBox installation?. Number of responses: 264 responses.](https://user-images.githubusercontent.com/13487278/113518949-6c80b600-9557-11eb-8788-4f6103b07033.png)

![Forms response chart. Question title: How many devices exist in your NetBox installation?. Number of responses: 274 responses.](https://user-images.githubusercontent.com/13487278/113518956-730f2d80-9557-11eb-98c0-d1355ced27d1.png)

It's awesome to see so many organizations using NetBox at scale!

![Forms response chart. Question title: How have you deployed NetBox?. Number of responses: 275 responses.](https://user-images.githubusercontent.com/13487278/113518960-786c7800-9557-11eb-9481-f5e44b752384.png)

There's been a slight uptick in the portion of organizations deploying NetBox as a container, though the traditional VM deployment still enjoys a healthy lead. Interestingly we're also seeing slightly higher bare metal deployment.

![Forms response chart. Question title: Which of the following have you deployed as multiple instances for high availability or load sharing (if any)?. Number of responses: 94 responses.](https://user-images.githubusercontent.com/13487278/113518964-7f938600-9557-11eb-8242-db96dc2fe8e7.png)

HA deployment numbers are up considerably across the board compared to last year.

![Forms response chart. Question title: How do you upgrade NetBox to a new release?. Number of responses: 270 responses.](https://user-images.githubusercontent.com/13487278/113518999-a782e980-9557-11eb-8ed4-409d24f41197.png)

Aside from the Docker portion increasing a few points, very little change here compared to last year.

![Forms response chart. Question title: What types of resources do you manage with NetBox?. Number of responses: 275 responses.](https://user-images.githubusercontent.com/13487278/113519003-b1a4e800-9557-11eb-9fbe-20682238eb30.png)

Again, not much change. Secrets usage has declined markedly and is due to be split off into a plugin for NetBox v2.12.

![Forms response chart. Question title: What NetBox features do you use?. Number of responses: 250 responses.](https://user-images.githubusercontent.com/13487278/113519011-b8cbf600-9557-11eb-97a7-0e6a9411edeb.png)

New on this year's survey is the plugins feature, used by nearly half of respondents.

![Forms response chart. Question title: What tools did you use before employing NetBox?. Number of responses: 265 responses.](https://user-images.githubusercontent.com/13487278/113519022-cd0ff300-9557-11eb-8c9e-7b9c09378e1a.png)

Strong evidence that Microsoft Excel has retained its position as the world's most popular IPAM software.

![Forms response chart. Question title: Which of the following tasks do you use NetBox to accomplish?. Number of responses: 266 responses.](https://user-images.githubusercontent.com/13487278/113519026-d39e6a80-9557-11eb-9680-586b043500fc.png)

The moderate upward trend across all integrations as compared to last year is encouraging.

![Forms response chart. Question title: How would you assess your or your network team's development abilities?. Number of responses: 271 responses.](https://user-images.githubusercontent.com/13487278/113519035-dc8f3c00-9557-11eb-85df-eab2f025200b.png)

Pretty similar responses to last year.

### Open-Ended Responses

Finally, we asked: If you could add one feature to NetBox, what would it be? Free-form answers are always challenging to quantify, but here are the most popular responses (those with at least three mentions), loosely correlated:

* VPN/overlay modeling (8)
* Firewall rule management (6)
* Floorplans (6)
* GraphQL (6)
* Network topology diagrams (5)
* BGP ASN modeling (5)
* Native single sign-on (SSO) support (4)
* Extended power modeling (3)
* Virtual circuits (3)

Overlay modeling is certainly something I'd like to implement, though it's going to be a huge undertaking and will require a great deal of planning to ensure we get the models right. Firewall rule management comes up pretty frequently, but it's so broad that IMO it easily warrants its own application, especially when you consider the myriad nuances of individual vendors and platforms.

Floor plan modeling was proposed in a very early feature request but was recently rejected as a core enhancement for the same reasons we ditched early attempts at native support for network diagramming: The complexity required to develop and maintain the feature significantly detracts from the core application, and its "ideal" implementation is bound to be highly user-dependent. This sort of functionality is much better suited as a plugin, and several NetBox plugins have been created to support network topology drawings.

GraphQL could make sense as a complement to the REST API, though it hasn't been a priority. This is another great candidate for initial development as a plugin. Once mature, it could be integrated with NetBox core pretty easily.

It's difficult to tell from a few terse responses exactly what people mean when they say "ASN modeling" or "BGP AS tracking," though it's clear the area deserves more investigation. I invite anyone who has a particular model or functionality in mind to submit a detailed feature request for further discussion.

There are [a lot](https://djangopackages.org/grids/g/authentication/) of external authentication apps for Django, and NetBox already supports customizable authentication backends. But I'm open to specific proposals by individuals who would like to contribute a more comprehensive implementation.

We have a few minor issues open relating to power modeling, though there's nothing super exciting. It was difficult to discern from the responses exactly what change(s) were being suggested, but I encourage anyone with a specific proposal to open a feature request.

And finally: virtual circuits. Yes. At some point. Much like overlay modeling, we need to consider all reasonable use cases and come up with a robust data model that can support all of them.

In addition to the suggestions above, I want to call out the following, all of which have already been addressed in the upcoming v2.11 release:

* Define rooms/locations within a site (#5895)
* Trace cable paths across circuits (#6071)
* Extend custom field support to additional models (#5370, #5401)
* Assign VLANs to multiple sites (#5284)
* Historical note tracking (#151)

Thanks again to everyone who participated, and keep an eye out for the v2.11 beta this week!

**Comment by tyler-8** [comment]
"Retail" is an industry that would be a sensible addition to the survey.

**Comment by eronlloyd** [comment]
Very insightful report. I'm surprised there wasn't more requests for DNS support, in particular.
